pkg/logql/rangemapper.go

```diff
@@ -0,0 +1,313 @@
+package logql
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+
+	"github.com/grafana/loki/pkg/logql/syntax"
+	util_log "github.com/grafana/loki/pkg/util/log"
+)
+
+type RangeVectorMapper struct {
```

Comment:
I'm a little late to the game so please excuse my ignorance. Why is this called a mapper? What does it do? Could you add some documentation?

Reply:
Since the rewriting follows the same pattern as in the [ShardMapper](https://github.com/grafana/loki/blob/main/pkg/logql/shardmapper.go#L96), we kept the same naming.

But thinking about it, `Vector` should not really be part of the name.